crates/system-services/src/deployer.rs

```diff
@@ -312,30 +313,99 @@ impl Deployer {
         }
     }
 
-    async fn deploy_system_spell(&self, spell_distro: SpellDistro) -> eyre::Result<ServiceStatus> {
+    async fn deploy_system_spell_common(
+        &self,
+        spell_distro: SpellDistro,
+    ) -> eyre::Result<ServiceStatus> {
         let spell_name = spell_distro.name.clone();
         match self.find_same_spell(&spell_distro) {
             ServiceUpdateStatus::NeedUpdate(spell_id) => {
                 tracing::debug!(
                     spell_name,
                     spell_id,
-                    "found existing spell that needs to be updated; will remove tha old spell and deploy a new one",
+                    "found existing spell that needs to be updated; will try to update script, trigger config and init data",
                 );
-                self.clean_old_spell(&spell_name, spell_id).await;
+                if let Err(err) = self.update_spell(&spell_distro, &spell_id).await {
+                    tracing::warn!(
+                        spell_id,
+                        spell_name,
+                        "can't update a spell (will remove the old spell and deploy a new one): {err}"
+                    );
+                    self.clean_old_spell(&spell_name, spell_id).await;
+                    let spell_id = self.deploy_spell_common(spell_distro).await?;
+                    tracing::info!(spell_name, spell_id, "redeployed a system spell",);
```

Comment:
```suggestion
                    tracing::info!(spell_name, spell_id, "redeployed a system spell");
```
Does log show `spell_name` and `spell_id`?

Reply:
yes